libc/src/__support/GPU/amdgpu/utils.h

```diff
@@ -21,10 +21,12 @@ namespace gpu {
 constexpr const uint64_t LANE_SIZE = __AMDGCN_WAVEFRONT_SIZE;
 
 /// Type aliases to the address spaces used by the AMDGPU backend.
-template <typename T> using Private = [[clang::opencl_private]] T;
-template <typename T> using Constant = [[clang::opencl_constant]] T;
-template <typename T> using Local = [[clang::opencl_local]] T;
-template <typename T> using Global = [[clang::opencl_global]] T;
+// FIXME: The OpenCL attributes are currently buggy on AMDGPU compared to the
```

Comment:
I don't think the asserts you are running into should be difficult to fix

Reply:
I have an issue open about one here https://github.com/llvm/llvm-project/issues/65069 already. This one might be related as well. I'm going to guess it's a lot of `LangOptions.OpenCL` causing this.